### PR TITLE
Schemas: Add ability to omit specific values

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds a menu item to omit a specific value from a form. Omitting the value disabled the property and removes its value from the from values. Enabling the property adds the original value back to the form values. 

Note: You can only omit an optional value

image
image

### CODE REVIEW

File: src/schemas/components/SchemaFormProperties.vue
Comment (pleek91) [inline src/schemas/components/SchemaFormProperties.vue:1]: There are a few updates around `null` vs `undefined`. `null` is 1-1 with python's `None` vs `undefined` is not a value and is removed from the form values. `undefined` values are removed as part of `JSON.stringify` anyway but also specifically removing them from the form value so keep that object consistent with what gets sent to the api.

File: src/schemas/components/SchemaFormProperty.vue [hunk 1]
```diff
@@ -64,7 +88,7 @@
       return null
     },
     set(value) {
-      emit('update:value', value)
+      emit('update:value', value ?? undefined)
```
Comment (pleek91) [inline src/schemas/components/SchemaFormProperty.vue:91]: Our core prefect-design components generally emit `null` for empty values. So making sure we coerce `null` values to `undefined` here.

File: src/schemas/components/SchemaFormPropertyMenu.vue [hunk 2]
```diff
@@ -15,6 +17,7 @@
 
   const props = defineProps<{
     kind: PrefectKind,
+    disabled?: boolean,
```
Comment (pleek91) [inline src/schemas/components/SchemaFormPropertyMenu.vue:20]: When a value is omitted you cannot change the prefect_kind. So adding the disabled prop hides all the kind options and just leaves the slot. Not super clean but works for now. Might revisit this if more things like this pop up.

Comment (znicholasbrown) [reply src/schemas/components/SchemaFormPropertyMenu.vue:20]: (non-blocking) - what does that look like? Just the label/description of the prop?

**Review by znicholasbrown — APPROVED** [review]

File: src/schemas/components/SchemaFormPropertyMenu.vue (hunk 2 above)
Comment (znicholasbrown) [reply src/schemas/components/SchemaFormPropertyMenu.vue:20]: scratch that I understand now - thought this was in the property component and not the menu

**Comment by znicholasbrown** [comment]
Not sure how I feel about the strikeout property names - they're a little hard to read and not entirely clear that it's related to the omitting. Would prefer subdued/disabled text with `(Omitted)` or something clearer but can handle in a follow up PR

**Comment by pleek91** [comment]
> Not sure how I feel about the strikeout property names - they're a little hard to read and not entirely clear that it's related to the omitting. Would prefer subdued/disabled text with `(Omitted)` or something clearer but can handle in a follow up PR

Agreed I think it could be improved. But we can tweak that at anytime so I figure stuff like that will get cleaned up toward the end when we're tweaking after trying everything together.

**Pull request merged**